e2e/nomostest/new.go

```diff
@@ -181,23 +181,23 @@ func SharedTestEnv(t nomostesting.NTB, opts *ntopts.New) *NT {
 
 	nt.detectGKEAutopilot(opts.SkipAutopilot)
 
-	// Print container logs in its own cleanup block to catch fatal errors from
-	// tests and test setup (including resetSyncedRepos).
 	t.Cleanup(func() {
 		if t.Failed() {
```

Comment:
Ah I was thinking the logic would look a bit differently here, such that we wouldn't print the logs again below if they were already printed here. With this implementation we could probably put this `printTestLogs` in a separate cleanup block, but I think this seems good. It seems reasonable to print the logs again if reset failed to help diagnose cleanup errors

Reply:
I thought you suggested putting them in the same cleanup block to make the control flow clearer.

The pupose of printing logs for both test failures and cleanup errors is to get the diagnosis in one shot. The reset should be resilient enough to proceed with the next test case even if the current one fails.